crates/hc_sandbox/tests/cli.rs

```diff
@@ -345,6 +349,99 @@ async fn generate_non_running_sandbox_and_call_list_dna() {
     assert!(exit_code.success());
 }
 
+/// Generates a sandbox and overwrites the conductor config with
+/// a specific allowed origin on the admin interface, then calls
+/// ListDna with the correct origin
+#[tokio::test(flavor = "multi_thread")]
+async fn generate_sandbox_and_call_list_dna_with_origin() {
+    clean_sandboxes().await;
+    package_fixture_if_not_packaged().await;
+
+    holochain_trace::test_run();
+    let mut cmd = get_sandbox_command();
+    cmd.env("RUST_BACKTRACE", "1")
+        .arg(format!(
+            "--holochain-path={}",
+            get_holochain_bin_path().to_str().unwrap()
+        ))
+        .arg("--piped")
+        .arg("generate")
+        .arg("--in-process-lair")
+        .arg("tests/fixtures/my-app/")
+        .stdin(Stdio::piped())
+        .stdout(Stdio::piped())
+        .stderr(Stdio::inherit())
+        .kill_on_drop(true);
+
+    let mut hc_admin = input_piped_password(&mut cmd).await;
+    let config_root_path = get_config_root_path(&mut hc_admin).await;
+    hc_admin.wait().await.unwrap();
+
+    // Overwrite the allowed_origins field
```

Comment:
I first thought that I could also add an option to `hc s generate` to specify the admin interface allowed origins, however I had to realize that this would lead to some problems with `hc zome-call` for example which requires the sandbox cli to make a connection to the admin websocket as well without necessarily knowing the proper origin. It would probably be possible to read that allowed origin from the config automatically but I decided to let that be for now. Open to be told otherwise.

Reply:
I think we probably want to allow setting an origin for app admin/app connections in the zome calls too but that's more complicated and not for this change. I think leaving that expecting no origins set is the right choice for this PR 👍🏻